### PR TITLE
Auto-deploy to environments on push

### PR DESCRIPTION
**Pull request by zuzak**

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -72,25 +55,33 @@ name: deploy
...
+      image: quay.io/ukhomeofficedigital/kd:latest
+      environment:
+        ENVIRONMENT: ${DRONE_STAGE_NAME}
+        VERSION: ${DRONE_COMMIT_SHA}
+        KUBE_TOKEN:
+          from_secret: ${DRONE_STAGE_NAME/-/_}
+      commands:
+        - cd kube
+        - ./deploy.sh
+			depends_on:
```
Comment (LiamMacP) [inline .drone.yml]: This required?

File: .drone.yml [hunk 2]
```diff
@@ -72,25 +55,33 @@ name: deploy
 
 trigger:
   event:
-    - promote
+    - push
```
Comment (LiamMacP) [inline .drone.yml:60]: This need a branch main restriction to?

**Comment by zuzak** [comment]
Parameter expansion is subject to pre-processing before the yaml is parsed, meaning I can't reference `${DRONE_STEP_NAME}` in the secret like I wanted to, so I've changed the behaviour to use a common Kubernetes token for every namespace. This won't work until we ask ACP to generate a token/update services in the ACP hub.

A future plan is to make a "safe" and a "dangerous" version of toolbox instead of porting the old psql-exec container. Then we can have two Kubernetes tokens for "safe" and "dangerous" namespaces, which I think would make more sense than having a different token for every namespace.

File: .drone.yml [hunk 3]
```diff
@@ -55,42 +39,43 @@ steps:
       tags:
         - build_${DRONE_BUILD_NUMBER}
         - ${DRONE_COMMIT_SHA}
+          - latest
```
Comment (LiamMacP) [inline .drone.yml:42]: Indentation on this wrong?

**Review by LiamMacP — APPROVED** [review]
Looks good, is this a candidate for the GitHub semver tagging actions too?

**Comment by zuzak** [comment]
> Looks good, is this a candidate for the GitHub semver tagging actions too?
I considered it, but I'm not sure it'd be helpful -- I expect we'll always be using `latest` everywhere.

Closing as this PR is part of #19 in a slightly different form.

**Pull request closed without merging**